base/lock.jl

```diff
@@ -484,7 +484,9 @@ end
 Create a level-triggered event source. Tasks that call [`wait`](@ref) on an
 `Event` are suspended and queued until [`notify`](@ref) is called on the `Event`.
 After `notify` is called, the `Event` remains in a signaled state and
-tasks will no longer block when waiting for it, until `reset` is called.
+tasks will no longer block when waiting for it, until [`reset(::Event)`](@ref)
+is called. Use [`isopen`](@ref) to check if it can still be signaled without
+having to call [`reset(::Event)`](@ref).
```

Comment:
I don't know what to is means, as an event can be signaled again even if not reset, and the wording seems to imply some sort of seq_cst behavior, which isn't something that multiple operations (check and reset) is valid terminology to have independently. The behavior when autoreset also seems contradictory to the statements here, as it talks about isopen being a property of writing (which is commonly true), but then associates it with reset, which is a reader operation and I don't think that synchronizes with writes, broadly speaking

Reply:
Maybe signaled is the wrong word, what I mean is that `isopen` can be used to check if the event will ever change from a un-set state to a set state again without having to explicitly `reset` it. Or put another way, if calling `notify` on the `Event` will have any effect.


> The behavior when autoreset also seems contradictory to the statements here, as it talks about isopen being a property of writing (which is commonly true), but then associates it with reset, which is a reader operation and I don't think that synchronizes with writes, broadly speaking

I'm not sure I quite understand what you mean, but I think the confusion might be because of the phrase "be signaled" coming across as 'you can signal this' instead of 'the state can change to signaled'. An alternative wording could be something like "Use `isopen` to check if it will ever block again while waiting without having to call `reset`"?

Reply:
"will have" sounds like a guarantee of some sort though, but it comes with so many asterisks (that it is not autoreset, that there is only one producer calling notify, and that the consumer does not call reset), I find the combination seems confusing how it would be used in practice